Feature: Setup poethepoet

antonykamp opened the pull request:
Fixes #68 

This PR installs `poethepoet` in order to run command line commands quickly.
It adds the command `poe format` to run `black` and `isort`.

You can find the new wiki entry here: https://github.com/nat-n/poethepoet

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [ ] ~Breaking changes anounced~
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

antonykamp committed:
dep: install poethepoet

Install poethepoet in order to add scripts to the pyproject.toml.

antonykamp committed:
feat: add command "poe format"

Add command "poe format" to run black and isort.

antonykamp committed:
doc: add link to internal poe documentation

SaturnHafen commented on pyproject.toml:
```diff
@@ -22,8 +22,23 @@ pytest = "^7.2.2"
+
+[tool.poe.tasks]
+
+    [tool.poe.tasks.format-isort]
+    help = "Run isort on the code base"
+    cmd  = "isort ."
+
+    [tool.poe.tasks.format-black]
+    help = "Run black on the code base"
+    cmd  = "black ."
+
+    [tool.poe.tasks.format]
+    help = "Run formating tools on the code base"
+    sequence  = ["format-isort", "format-black"]
```
Is this indentation necessary?

antonykamp replied:
it's common practice as you see here: https://github.com/nat-n/poethepoet/blob/5bc9d72436c3ca582d326bfa06fc137df086bafb/pyproject.toml#LL43C7-L43C7

And it's in my opinion more readable 😇

SaturnHafen commented on pyproject.toml:
```diff
@@ -22,8 +22,23 @@ pytest = "^7.2.2"
+
+[tool.poe.tasks]
+
+    [tool.poe.tasks.format-isort]
+    help = "Run isort on the code base"
+    cmd  = "isort ."
+
+    [tool.poe.tasks.format-black]
+    help = "Run black on the code base"
+    cmd  = "black ."
+
+    [tool.poe.tasks.format]
+    help = "Run formating tools on the code base"
+    sequence  = ["format-isort", "format-black"]
```
We could probably run `pylint` as well, couldn't we?

antonykamp replied:
I wouldn't add `pylint` to the `format` command, because a linter and a formatter are two different things 😇 We could add a new command `prepare-push` and run `poe format` and `pylint`.

SaturnHafen reviewed (approved):
I like it so far :+1: 
I personally would like to have all the commands that are executed in the ci (formatter, tests & linter) as a single poe shortcut.

antonykamp commented:
> I personally would like to have all the commands that are executed in the ci (formatter, tests & linter) as a single poe shortcut.

Good idea! I'll add some more commands! 🥳

antonykamp committed:
feat: add commands lint and test

Add commands to poe to run linting and testing frameworks.

antonykamp committed:
fix: use poe commands in github action

antonykamp committed:
fix(ci): call poe via poetry run

You have to call poe via poetry run in order to find the dependency poethepoet.

antonykamp committed:
fix: typo

tests -> test

instant-sky reviewed (approved):
Looks very promising! I am really looking forward to use these commands 🥳
